chips/nrf52/src/spi.rs

```diff
@@ -0,0 +1,501 @@
+pub static mut SPIM2: SPIM = SPIM::new(2);
+
+mod registers {
+    pub mod spim {
+        #![allow(dead_code)]
+        use kernel::common::VolatileCell;
+        use nrf5x::peripheral_interrupts::NvicIdx;
+        use nrf5x::pinmux::Pinmux;
+
+        pub const INSTANCES: [(*const SPIM, NvicIdx); 3] =
+            [(0x40003000 as *const SPIM, NvicIdx::SPI0_TWI0),
+             (0x40004000 as *const SPIM, NvicIdx::SPI1_TWI1),
+             (0x40023000 as *const SPIM, NvicIdx::SPIM2_SPIS2_SPI2)];
+
+        bitfield!{
+            // Represents bitfields in INTENSET and INTENCLR registers.
+            #[derive(Copy, Clone)]
+            pub struct InterruptEnable(u32);
+            impl Debug;
+            pub stopped, set_stopped:  1,  1;
+            pub end_rx,  set_end_rx:   4,  4;
+            pub end,     set_end:      6,  6;
+            pub end_tx,  set_end_tx:   8,  8;
+            pub started, set_started: 19, 19;
+        }
+
+        bitfield!{
+            // Represents bitfields in CONFIG register.
+            #[derive(Copy, Clone)]
+            pub struct Config(u32);
+            impl Debug;
+            pub order,          set_order:          0,  0;
+            pub clock_phase,    set_clock_phase:    1,  1;
+            pub clock_polarity, set_clock_polarity: 2,  2;
+        }
+
+        #[repr(u32)]
+        #[derive(Copy, Clone)]
+        pub enum Frequency {
+            K125 = 0x02000000,
+            K250 = 0x04000000,
+            K500 = 0x08000000,
+            M1 = 0x10000000,
+            M2 = 0x20000000,
+            M4 = 0x40000000,
+            M8 = 0x80000000,
+        }
+
+        impl From<Frequency> for u32 {
+            fn from(freq: Frequency) -> u32 {
+                match freq {
+                    Frequency::K125 => 125_000,
+                    Frequency::K250 => 250_000,
+                    Frequency::K500 => 500_000,
+                    Frequency::M1 => 1_000_000,
+                    Frequency::M2 => 2_000_000,
+                    Frequency::M4 => 4_000_000,
+                    Frequency::M8 => 8_000_000,
+                }
+            }
+        }
+
+        impl From<u32> for Frequency {
+            fn from(freq: u32) -> Frequency {
+                if freq < 250_000 {
+                    Frequency::K125
+                } else if freq < 500_000 {
+                    Frequency::K250
+                } else if freq < 1_000_000 {
+                    Frequency::K500
+                } else if freq < 2_000_000 {
+                    Frequency::M1
+                } else if freq < 4_000_000 {
+                    Frequency::M2
+                } else if freq < 8_000_000 {
+                    Frequency::M4
+                } else {
+                    Frequency::M8
+                }
+            }
+        }
+
+        #[repr(u32)]
+        #[derive(Copy, Clone)]
+        pub enum Enable {
+            Disabled = 0,
+            Enabled = 7,
+        }
+
+        #[allow(non_snake_case)]
```

Comment:
Seems like this is unnecessary since all is already snake_cases?

Reply:
Good catch. I started out by copy-pasting some other code as a template and that must have came with it.